app/graphql/types/list_albums.rb

```diff
@@ -0,0 +1,55 @@
+require 'search_object/plugin/graphql'
+
+module Types
+  class ListAlbums < QueryType
+    include SearchObject.module(:graphql)
+
+    scope { AlbumsService.new(ENDPOINT).list_all_albums }
+
+    type types[Types::AlbumType]
+
+    class AlbumsSort < ::Types::BaseInputObject
+      FieldEnum = GraphQL::EnumType.define do
+        name 'Fields'
+        description 'fields by which you can sort'
+        value('TITLE', 'Title of album', value: 'title')
+        value('USER_ID', 'User id of album owner', value: 'user_id')
+        value('ID', 'Id of album', value: 'id')
+      end
+
+      OrderEnum = GraphQL::EnumType.define do
+        name 'Order'
+        description 'Sort by'
+        value('ASC', 'Ascending order', value: 0)
+        value('DESC', 'Descending order', value: 1)
+      end
+
+      argument :order, OrderEnum, required: false, description: 'Sort by title or user id'
+      argument :field, FieldEnum, required: false, description: 'Sort by title or user id'
+    end
+
+    option :offset, type: types.Int, with: :apply_offset
```

Comment:
I'm not so sure but if this `option` methods are applied sequentially, this means you are doing pagination first and applying the filter later, which makes no sense because after doing filtering the quantity of elements it's likely to be reduced.